### DISCUSSION TITLE
planet positions in ecliptic coordinates of date

### DISCUSSION THREAD

**Discussion by GernotSchreider** [cosinekitty/astronomy#318] [Q&A]
Hi 
I am trying to get my head around the proper usage of functions but I guess I could use some help

I want to calculate the position of planets for a given date and time and the result should be in ecliptic coordinates of date
I could not figure out how to properly convert equatorial coordinates, which are returned by Equator() to ecliptic.
Thanks

**Comment by cosinekitty** [comment]
Which programming language are you using?

**Reply by GernotSchreider** [reply]
Thanks for this quick reply, I am using Python

**Comment by cosinekitty** [comment]
By the way, welcome to the project, @GernotSchreider!  I just created the demo program [ecliptic_of_date.py](https://github.com/cosinekitty/astronomy/blob/master/demo/python/ecliptic_of_date.py) that shows how to do this. The basic idea is to use the vector part of the return value from `Equator`, called `vec`. Note that I call `Equator` with the `ofdate` parameter set to `True`, which causes it to return equatorial coordinates of date. Then I use a rotation matrix to convert the vector to true ecliptic coordinates of date. Finally, I convert the ecliptic vector into ecliptic spherical coordinates and print them. I hope this helps!

**Reply by GernotSchreider** [reply]
Thank you Don, 
this perfectly explains things now for my problem at hand.
I was missing the transformation from cartesian to spherical coords for the vector.
I had tried to understand the code and the general approach you have but was somehow lost.

I was starting to build my own implementation of VSOP calculation when I found your repository. So thank you for sharing it, it is great.


I still see some minor delta when comparing results to Stellarium, esp. for Mercury, Venus and Mars. For today around noon CEST it amounts to some 20'' in longitude (eclip to date). For Jupiter and Saturn it looks much better. 
Stellarium uses DE430/431 models, which apparently is a different approach to VSOP87. Does the difference in models explain the delta I am seeing or do I miss something else here?

BTW I should mention my current pet project I am working on
I am building a graphical simulation of an astronomical clock, based on a real clock.
It shows time of local timezone and the true solar time on a 24h clock face, furthermore a circle displaying the ecliptic +/- a few degrees with the major stars, which is also moving in real time and then some pointers to the position of the planets.
This is also my pet project to learn python ;-)

**Comment by cosinekitty** [comment]
> Does the difference in models explain the delta I am seeing or do I miss something else here?

No, both VSOP87 and DE431 will give you extremely close results. They will differ by small fractions of an arcsecond. That kind of discrepancy would matter for professional astronomers only. Astronomy Engine uses **truncated** VSOP87 series for smaller code and much faster performance, at the expense of approximately one arcminute of error. That is consistent with the 20 arcseconds difference you mention.

This was an intentional design decision to fill a niche I saw in astronomy calculation software: there was a gap between extremely simple but very inaccurate formulas (more than 1 degree of error in some cases!), and hyper-accurate formulas (milli-arcseconds) that were excessively large, complicated, and slow to calculate.

Astronomy Engine's goal is to be accurate enough for someone using a backyard telescope, but small and fast enough to work in a web browser's JavaScript engine or a microcontroller's limited RAM. It is an ideal trade-off for the vast majority of amateur/civilian astronomy uses.

> I am building a graphical simulation of an astronomical clock, based on a real clock.

That sounds like a fantastic project!  I would love to see what you come up with.

**Reply by GernotSchreider** [reply]
Don, that makes a lot of sense and explains the deltas. I was kind of expecting that both models should provide sub-arcsecond precision. 
The precision is sufficient for my current purpose. I was just trying to understand if there is somewhere a problem, which I did not spot.
I can share some result when available.
Cheers
Gernot
